packages/generator-joplin/generators/app/templates/api/types.ts

```diff
@@ -557,17 +581,19 @@ export interface CodeMirrorControl {
 		 *
 		 * Using `autocompletion({ override: [ ... ]})` causes errors when done by multiple plugins.
 		 */
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
 		completionSource(completionSource: any): any;
 
 		/**
 		 * Creates an extension that enables or disables [`languageData`-based autocompletion](https://codemirror.net/docs/ref/#autocomplete.autocompletion^config.override).
 		 */
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
 		enableLanguageDataAutocomplete: { of: (enabled: boolean)=> any };
 	};
 }
 
-export interface CodeMirrorContentScriptModule extends Omit<ContentScriptModule, 'plugin'> {
-	plugin: (codeMirrorControl: CodeMirrorControl)=> void;
+export interface MarkdownEditorContentScriptModule extends Omit<ContentScriptModule, 'plugin'> {
+	plugin: (editorControl: CodeMirrorControl)=> void;
 }
```

Comment:
The [pull request (#10015)](https://github.com/laurent22/joplin/pull/10015) that renamed `CodeMirrorContentScriptModule` to `MarkdownEditorContentScriptModule` was originally accepted because it came shortly after the introduction of the `CodeMirrorContentScriptModule` type. However, it didn't update `generator-joplin`.

The updated generator types (with `CodeMirrorContentScriptModule`) have yet to be released to NPM. Thus, it should still be safe to apply this renaming to `generator-joplin` now.